src/NewProject/ProjectManager/project_manager.cpp

```diff
@@ -1716,7 +1716,100 @@ int ProjectManager::CreateAndAddFile(const QString& suffix,
 void ProjectManager::UpdateProjectInternal(const ProjectOptions& opt,
                                            bool setTargetConstr) {
   setCurrentFileSet(opt.currentFileSet);
-  const QList<filedata> listFile = opt.sourceFileData.fileData;
+  auto addDesignFiles = [this](const QString& commands, const QString& libs,
+                               const QString& fileNames, int lang,
+                               const QString& grName, bool isFileCopy,
+                               bool localToProject) {
+    setDesignFiles(commands, libs, fileNames, lang, grName, isFileCopy,
+                   localToProject);
+  };
+
+  AddFiles(opt.sourceFileData, addDesignFiles);
```

Comment:
What went into upstream testing with this refactoring? It's not available in the PR description.

Reply:
this change covered by unit testing. And there are actually not much changes. Just some code was moved to separate function.